src/inbox.rs

```diff
@@ -293,6 +231,80 @@ impl<A> Chan<A> {
     }
 }
 
+pub trait TrySend<M> {
+    fn try_send(&self, message: M) -> Result<Result<(), MailboxFull<M>>, Error>;
```

Comment:
Isn't this essentially equivalent to what we had before, but based on a trait rather than an enum?

Reply:
It is functionally equivalent but a trait with type parameters is a compile-time concept and an enum is a runtime-concept.

Reply:
We could not have a trait by duplicating all of the `send_future::Sending` machinery but I figured that is too much code duplication.